kore/src/Kore/Step/Simplification/In.hs

```diff
@@ -81,9 +82,13 @@ simplifyEvaluatedIn sideCondition first second
   | OrPattern.isTrue first = Ceil.simplifyEvaluated sideCondition second
   | OrPattern.isTrue second = Ceil.simplifyEvaluated sideCondition first
 
-  | otherwise =
-    OrPattern.flatten <$> sequence
-                            (makeEvaluateIn sideCondition <$> first <*> second)
+  | otherwise = do
+    result <-
+        OrPattern.observeAllT $ do
+            pattFirst <- Logic.scatter first
+            pattSecond <- Logic.scatter second
+            makeEvaluateIn sideCondition pattFirst pattSecond
+    return (OrPattern.flatten result)
```

Comment:
```suggestion
  | otherwise =
    OrPattern.observeAllT $ do
        pattFirst <- Logic.scatter first
        pattSecond <- Logic.scatter second
        makeEvaluateIn sideCondition pattFirst pattSecond >>= Logic.scatter
```